src/DrawManager.sol

```diff
@@ -209,28 +218,51 @@ contract DrawManager {
     uint24 drawId = prizePool.getDrawIdToAward(); 
     uint48 closesAt = prizePool.drawClosesAt(drawId);
     if (closesAt > block.timestamp) revert DrawHasNotClosed();
-    StartDrawAuction memory lastRequest = _lastStartDrawAuction;
-    if (lastRequest.drawId == drawId) revert AlreadyStartedDraw();
     if (rng.requestedAtBlock(_rngRequestId) != block.number) revert RngRequestNotInSameBlock();
+    
+    StartDrawAuction memory lastRequest = getLastStartDrawAuction();
+    uint auctionOpenedAt;
+    
+    if (lastRequest.drawId != drawId) { // if this request is for a new draw
+      // auctioned opened at the close of the draw
+      auctionOpenedAt = closesAt;
+      // clear out the old ones
+      while (_startDrawAuctions.length > 0) {
+        _startDrawAuctions.pop();
+      }
+    } else { // the old request is for the same draw
+      if (!rng.isRequestFailed(lastRequest.rngRequestId)) { // if the request failed
+        revert AlreadyStartedDraw();
+      } else if (_startDrawAuctions.length == maxRetries) { // if request has failed and we have retried too many times
+        revert RetryLimitReached();
+      } else if (_rngRequestId <= lastRequest.rngRequestId) { // if the request has failed, we haven't retried too many times, but the rng request has been used
```

Comment:
We are making assumptions that the RNG request ID is strictly increasing, which may not be the case with other RNG sources. We should be using the `rng.requestedAtBlock` function to check the relative block numbers instead.